database/methods.js

```diff
@@ -4,20 +4,38 @@ const driver = neo4j.driver("bolt://localhost:7687", neo4j.auth.basic("neo4j", "
 
 const getSong = (song_id, res) => {
   var session = driver.session();
-  const cypher = 'MATCH (song:Song {id: {id} }) RETURN song';
   const params = {id: song_id};
-  console.log(params);
-  return session
-    .run(cypher, params)
+  
+  return Promise.all([
+    session.run('MATCH (song:Song {id: {id} }) RETURN song.name', params),
+    session.run('MATCH (song:Song {id: {id} })-[:APPEARS_ON]->(album:Album) RETURN album', params)
+  ])
+  .then(result => {
+    res.send(result);
+    session.close();
+  })
+  .catch(error => {
+    session.close();
+    throw error;
+  });
+}
+
+const postSong = (song_data, res) => {
+  var session = driver.session();
+  const cypherCreateSong = 'CREATE (song:Song {name: {songname}, art: {songart}, year: {songyear}})';
+  const cypherCreateSongParams = {songname: song_data.songname, songart: song_data.songart, songyear: song_data.songyear};
+
+  session.run(cypherCreateSong, cypherCreateSongParams)
     .then(result => {
-      session.close();
-      console.log(result);
-      res.send(result.records);
+      res.status(200);
+      res.end();
     })
     .catch(error => {
-      session.close();
+      res.status(500);
+      res.end();
       throw error;
     });
-}
+  }
```

Comment:
why did you remove the session.close() statements? (I notice you have them in the read function)

Reply:
Good catch, I should've left that in.